mttl/models/containers/selectors/base_selectors.py

```diff
@@ -461,9 +492,9 @@ def forward(self, input, **kwargs) -> BatchSequenceExpertsAndWeightsSelectorOutp
             # soft routing
             selected_experts = SelectorOutput.ALL_EXPERTS
 
-        g = self.info_container.get("routing_gates", [])
+        g = getattr(self.info_container, "routing_gates", [])
```

Comment:
Not really introduced in your PR but would initializing `info_container.routing_gates = []` in the class __init__ work and be cleaner?

Reply:
which class init? InfoContainer?